### ISSUE TITLE
Size for 0d (scalar) variable is float

### ISSUE THREAD

**Issue by gerritholl** [Unidata/netcdf4-python#957]
When a variable is 0-dimensional, its `.size` is a float `1.0` where it should be an int `1`:

```python
import netCDF4
ds = netCDF4.Dataset("/tmp/foo.nc", "w")
v = ds.createVariable("spam", "uint16", dimensions=())
print(v.size)
```

Gives:

```
1.0
```

Suse Linux 15.0.  Libraries netcdf4 1.5.0.1 and libnetcdf 4.6.2 installed from conda-forge.

**Comment by jswhit** [comment]
This is because, for some reason `numpy.prod(())=1.0`.  Fix is in #958

**Comment by jswhit** [comment]
Merged PR #957
